src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1344,13 +1345,22 @@ private bool OnlyWhitespaceOrCommentsAfterExtent(InternalScriptExtent extent)
 
         internal bool IsPipeContinuance(IScriptExtent extent)
         {
-            return extent.EndOffset < _script.Length && PipeContinuanceAfterExtent(extent);
+            // If the first non-whitespace & non-comment (regular or block) character following a newline is a pipe, we have
+            // pipe continuance.
+            return extent.EndOffset < _script.Length &&
+                   ContinuanceAfterExtent(extent, continuanceChar: '|', skipComment: true);
         }
 
-        private bool PipeContinuanceAfterExtent(IScriptExtent extent)
+        internal bool IsTernaryContinuance(IScriptExtent extent)
+        {
+            // If the first non-whitespace character following a newline is a question mark, we have
+            // ternary continuance
+            return extent.EndOffset < _script.Length &&
+                   ContinuanceAfterExtent(extent, continuanceChar: '?', skipComment: false);
```

Comment:
>skipComment: false [](start = 72, length = 18)

Why disallow comments?

Reply:
I thought it was not necessary to allow for comments. I changed my mind, as comments are allowed before `:`.
Will change this.

Reply:
I'd like comment support in this, but without blank lines, e.g.

```powershell
$something -eq $someValue
    # Describe what this does
    ? Do-ThisThing
    # Describe what the alternative does
    : Do-ThatThing
```

Reply:
Updated to allow comments.

Reply:
Introducing new look-ahead could hurt parsing performance so be sure to measure the impact.

Reply:
Will measure the impact and reply back.